Core/Core/Configuration/Combine/Config/Config.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  Config.swift
+//  Core
+//
+//  Created by Muhammad Umer on 11/11/2023.
+//
+
+import Foundation
+
+public protocol ConfigProtocol {
+    var baseURL: URL { get }
+    var oAuthClientId: String { get }
+    var tokenType: TokenType { get }
+    var feedbackEmail: String { get }
+    var appStoreLink: String { get }
+    var agreement: AgreementConfig { get }
+    var firebase: FirebaseConfig { get }
+    var features: FeaturesConfig { get }
+}
+
+public enum TokenType: String {
+    case jwt = "JWT"
+    case bearer = "BEARER"
+}
+
+private enum ConfigKeys: String {
+    case baseURL = "API_HOST_URL"
+    case oAuthClientID = "OAUTH_CLIENT_ID"
+    case tokenType = "TOKEN_TYPE"
+    case feedbackEmailAddress = "FEEDBACK_EMAIL_ADDRESS"
+    case environmentDisplayName = "ENVIRONMENT_DISPLAY_NAME"
+    case platformName = "PLATFORM_NAME"
+    case organizationCode = "ORGANIZATION_CODE"
+    case appstoreID = "APP_STORE_ID"
+}
+
+public class Config {
+    public static let shared = Config()
+    
+    internal var properties: [String: Any] = [:]
+    
+    internal init(properties: [String: Any] = [:]) {
+        self.properties = properties
+    }
+    
+    internal convenience init() {
+        self.init(properties: [:])
+        loadAndParseConfig()
+    }
+    
+    private func loadAndParseConfig() {
+        guard let path = Bundle.main.path(forResource: "config", ofType: "plist"),
+              let data = try? Data(contentsOf: URL(fileURLWithPath: path)),
+              let dict = try? PropertyListSerialization.propertyList(from: data, options: [], format: nil) as? [String: Any] else { return }
+        
+        properties = dict
+    }
+    
+    internal subscript(key: String) -> Any? {
+        return properties[key]
+    }
+    
+    func dict(for key: String) -> [String: Any]? {
+        return properties[key] as? [String: Any]
+    }
+    
+    func value<T>(for key: String) -> T? {
+        return properties[key] as? T
+    }
+    
+    func value(for key: String) -> Any? {
+        return properties[key]
+    }
+    
+    func value(for key: String, dict: [String: Any]) -> String? {
+        return dict[key] as? String ?? nil
+    }
+    
+    func string(for key: String) -> String? {
+        return value(for: key) as? String ?? nil
+    }
+    
+    func string(for key: String, dict: [String: Any]) -> String? {
+        return value(for: key, dict: dict)
+    }
+    
+    func bool(for key: String) -> Bool {
+        return value(for: key) as? Bool ?? false
+    }
+}
+
+extension Config: ConfigProtocol {
+    public var baseURL: URL {
+        return URL(string: string(for: ConfigKeys.baseURL.rawValue)!)!
```

Comment:
I understand baseURL is important part and we need to have this
maybe it can be something like
```    
    public var baseURL: URL {
        if let string = string(for: ConfigKeys.baseURL2.rawValue),
            let url = URL(string: string) {
            return url
        }
        fatalError("Can't find baseURL in Config!")
    }
```

Reply:
i used convention that was being followed in project, i think we can use guard let here and return `URL(string "")` instead of doing `fataError`, what are your thoughts?

Reply:
i am not in favour of `fataError()`, i think app should handle issue gracefully.

Reply:
@mumer92 code we have now (with force unwrapping) will just crash if baseURL is not present in config.
URL(string: "") also returns nil and will crash with force unwrapping
I understand app can't work without baseURL - we need to try to show what is the problem exactly (fatalError show a message at least). It could be some Log etc. Just provide issue description.
I'm open to any options without force unwrapping.
What do you think?

Reply:
what are your thoughts on using:
```
public var baseURL: URL {
        guard let urlString = string(for: ConfigKeys.baseURL.rawValue),
              let url = URL(string: urlString) else {
            return URL(string: "")!
        }
        return url
    }
```

i believe we can do the same with `oAuthClientId`

Reply:
URL(string: "")! produces crash
why this Crash is better than ?Crash with message"? )

Reply:
alright i will add `fatalError("Can't find baseURL in Config!")`, in the mean time you can continue with other changes for the review.

Reply:
all other looks good for me. good work @mumer92 !

Reply:
@rnr i have updated it. please have a look.